File: javascript/line-up/line-up.js
//
// This is only a SKELETON file for the 'Line Up' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const format = (name, number) => {
  let numberStr = number.toString();
  let lastDigit = numberStr[numberStr.length - 1];
  let result = "";

  switch (lastDigit) {
    case "1":
      if (numberStr.length >= 2 && numberStr[numberStr.length - 2] === "1") {
        result += `${name}, you are the ${numberStr}th customer we serve today. Thank you!`;
        break;
      }
      result += `${name}, you are the ${numberStr}st customer we serve today. Thank you!`;
      break;
    case "2":
      if (numberStr.length >= 2 && numberStr[numberStr.length - 2] === "1") {
        result += `${name}, you are the ${numberStr}th customer we serve today. Thank you!`;
        break;
      }
      result += `${name}, you are the ${numberStr}nd customer we serve today. Thank you!`;
      break;
    case "3":
      if (numberStr.length >= 2 && numberStr[numberStr.length - 2] === "1") {
        result += `${name}, you are the ${numberStr}th customer we serve today. Thank you!`;
        break;
      }
      result += `${name}, you are the ${numberStr}rd customer we serve today. Thank you!`;
      break;
    default:
      result += `${name}, you are the ${numberStr}th customer we serve today. Thank you!`;
  }

  return result;
};

File: python/line-up/line_up.py
"""Module for generating a thank you message for a customer based on their position in line.
"""
def line_up(name, number):
    """Function to generate a thank you message for a customer based on their position in line.

    Args:
        name (str): The name of the customer.
        number (int): The position of the customer in line.

    Returns:
        str: A thank you message for the customer.
    """
    number_suffix = ""

    if str(number).endswith("11") or str(number).endswith("12") or str(number).endswith("13"):
        number_suffix = "th"
        thanks_sentence = f"{name}, you are the {number}{number_suffix} customer we serve today. Thank you!"
        return thanks_sentence

    if str(number).endswith("1"):
        number_suffix = "st"
    elif str(number).endswith("2"):
        number_suffix = "nd"
    elif str(number).endswith("3"):
        number_suffix = "rd"
    else:
        number_suffix = "th"

    thanks_sentence = f"{name}, you are the {number}{number_suffix} customer we serve today. Thank you!"
    return thanks_sentence
